server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/resources/cucumber/KeywordsConsistencyOnDelegationMailbox.feature

```diff
@@ -0,0 +1,190 @@
+#                                                              *
+# Unless required by applicable law or agreed to in writing,   *
+# software distributed under the License is distributed on an  *
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+# KIND, either express or implied.  See the License for the    *
+# specific language governing permissions and limitations      *
+# under the License.                                           *
+# **************************************************************/
+
+Feature: JMAP keywords consistency on delegation mailbox
+
+  Background:
+    Given a domain named "domain.tld"
+    And a user "alice@domain.tld"
+    And a user "bob@domain.tld"
+    And "alice@domain.tld" has a mailbox "notShared"
+    And "alice@domain.tld" has a mailbox "shared"
+    And "alice@domain.tld" has a mailbox "trash"
+    And "alice@domain.tld" shares its mailbox with write right "shared" with "bob@domain.tld"
+    And "alice@domain.tld" is connected
+    And the user has a message "mess" in "notShared" mailbox with subject "My awesome subject", content "This is the content"
+    And the user copy "mess" from mailbox "notShared" to mailbox "shared"
+
+  Scenario: getMessageList filtered by flag should combine flag when delegation mailbox
+    Given "bob@domain.tld" is connected
+    And the user set flags on "mess" to "$Flagged"
+    When "alice@domain.tld" is connected
+    And the user asks for message list in mailboxes "shared,notShared" with flag "$Flagged"
+    Then the message list has size 1
+    And the message list contains "mess"
+
+  Scenario: getMessageList filtered by flag should keep flag on delegation mailbox
+    Given "bob@domain.tld" is connected
+    And the user set flags on "mess" to "$Flagged"
+    When "alice@domain.tld" is connected
+    And the user asks for message list in mailboxes "notShared" with flag "$Flagged"
+    Then the message list is empty
+
+  Scenario: getMessageList filtered by flag should keep flag on non-shared mailbox
+    Given "bob@domain.tld" is connected
+    And the user set flags on "mess" to "$Flagged"
+    When "alice@domain.tld" is connected
+    And the user asks for message list in mailboxes "shared" with flag "$Flagged"
+    Then the message list has size 1
+    And the message list contains "mess"
+
+  @Ignore
+  Scenario: (Does not work) Get message list should insection Draft flag on all mailbox
+    Given "bob@domain.tld" is connected
+    And the user set flags $Draft via messageIdProbe of "mess" on "shared" of user "alice@domain.tld"
+    When "alice@domain.tld" is connected
+    And the user asks for message list in mailboxes "shared,notShared" with flag "$Draft"
+    Then the message list is empty
+
+  Scenario: Get message list should keep flags on non-shared mailbox
+    Given "bob@domain.tld" is connected
+    And the user set flags $Draft via messageIdProbe of "mess" on "shared" of user "alice@domain.tld"
```

Comment:
Now, 
- There is error when we update $Draft flag via IMAP with delegation mailbox (It does not support for now)
- We do not allow update $Draft flag via JMAP
--> That's why I chose probe for updating $Draft flag on message